src/db/models/Integrations.js

```diff
@@ -1,23 +1,315 @@
+    return this.createIntegration({
+      name,
+      brandId,
+      kind: KIND_CHOICES.MESSENGER,
+    });
+  }
+
+  /**
+   * Update messenger integration document
+   * @param {Object} object - Integration main doc object
+   * @param {string} object.name - Integration name
+   * @param {string} object.brandId - Integration brand id
+   * @return {Promise} return null promise
+   */
+  static updateMessengerIntegration(_id, { name, brandId }) {
+    return this.update({ _id }, { $set: { name, brandId } }, { runValidators: true });
+  }
+
+  /**
+   * Save messenger appearance data
+   * @param {string} _id
+   * @param {Object} object - MessengerUiOptions object TODO: need more elaborate documentation
+   * @param {string} object.color - MessengerUiOptions color TODO: need more elaborate documentation
+   * @param {string} object.wallpaper - MessengerUiOptions wallpaper
+   * @param {string} object.logo - Messenger logo TODO: need more elaborate documentation
+   * @return {Promise} returns null promise
+   */
+  static saveMessengerAppearanceData(_id, { color, wallpaper, logo }) {
+    return this.update(
+      { _id },
+      { $set: { uiOptions: { color, wallpaper, logo } } },
+      { runValdatiors: true },
+    );
+  }
+
+  /**
+  * Saves messenger data to integration document
+  * @param {Object} doc.messengerData - MessengerData object
+  * @param {Boolean} doc.messengerData.notifyCustomer - Identicates whether
+  *    customer should be notified or not TODO: need more elaborate documentation
+  * @param {string} doc.messengerData.availabilityMethod - Sets messenger
+  *    availability method as auto or manual TODO: need more elaborate documentation
+  * @param {Boolean} doc.messengerData.isOnline - Identicates whether messenger in online or not
+  * @param {Object[]} doc.messengerData.onlineHours - OnlineHours object array
+  * @param {string} doc.messengerData.onlineHours.day - OnlineHours day
+  * @param {string} doc.messengerData.onlineHours.from - OnlineHours from
+  * @param {string} doc.messengerData.onlineHours.to  - OnlineHours to
+  * @param {string} doc.messengerData.timezone - Timezone
+  * @param {string} doc.messengerData.welcomeMessage - Message displayed on welcome
+  *    TODO: need more elaborate documentation
+  * @param {string} doc.messengerData.awayMessage - Message displayed when status becomes away
+  *    TODO: need more elaborate documentation
+  * @param {string} doc.messengerData.thankYouMessage - Thank you message
+  *    TODO: need more elaborate documentation
+  * @param {string} doc.messengerData.uiOptions.color - Color of messenger
+  * @param {string} doc.messengerData.uiOptions.wallpaper - Wallpaper image for messenger
+  * @param {string} doc.messengerData.uiOptions.logo - Logo used in the embedded messenger
+  * @return {Promise} returns null promise
+  */
+  static saveMessengerConfigs(_id, messengerData) {
+    return this.update({ _id }, { $set: { messengerData } }, { runValidators: true });
```

Comment:
return updated integration
await this.update ....
return this.findOne({ _id })